### PR TITLE
fix: upgrade zoe sketchbook for new ERTP and Zoe APIs.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
copied in all new contracts from agoric-sdk.
updated package.json

https://github.com/Agoric/agoric-sdk/issues/716

### CODE REVIEW

File: examples/baytownBucks.js [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
+import produceIssuer from '@agoric/ertp';
 
 // Let's create a community currency and make a payment for Alice.
 // This example uses ERTP but does not use Zoe, since no offers are
 // being made.
 
-// First, let's create a mint
-const baytownBucksMint = makeMint('BaytownBucks');
+// First, let's create an issuer
+export const baytownBucksBundle = produceIssuer('BaytownBucks');
```
Comment (katelynsills) [inline examples/baytownBucks.js]: I like the name `bundle` for this!

Comment (katelynsills) [reply examples/baytownBucks.js]: Should we export the entire bundle, including the mint? Even though this is only a test example, maybe we could use it as a teaching tool, that the mint is a special authority?

Comment (katelynsills) [reply examples/baytownBucks.js]: The previous example only exported the issuer

File: examples/baytownBucks.js [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
-// Note: exported for testing purposes
-export const baytownBucksAssay = baytownBucksMint.getAssay();
+// Now let's get the mint from the issuer
+const baytownBucksMint = baytownBucksBundle.mint;
 
-// Let's give ourselves a easy way of labeling a number as baytownBucks.
+// Let's give ourselves a way to create amounts of baytownBucks.
 // Note: exported for testing purposes
-export const baytownBucks = baytownBucksAssay.makeUnits;
+export const baytownBucks = baytownBucksBundle.amountMath;
```
Comment (katelynsills) [inline examples/baytownBucks.js]:
```suggestion
export const baytownBucks = baytownBucksBundle.amountMath.make;
```
I think the entire amountMath shouldn't be called `baytownBucks` - just the `make` part

File: examples/baytownBucks.js [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
+
+// Now let's create a payment to hold our community treasury funds.
+const payment = baytownBucksMint.mintPayment(
+  baytownBucks.make(1000),
+  'community treasury',
```
Comment (katelynsills) [inline examples/baytownBucks.js]: We don't have any payment names/memos any longer

File: examples/baytownBucks.js [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
-// Now let's create a purse to hold our community treasury funds.
-const purse = baytownBucksMint.mint(baytownBucks(1000), 'community treasury');
+const purse = baytownBucksBundle.issuer.makeEmptyPurse();
+purse.deposit(payment);
 
 // Let's make that payment for Alice.
 // Note: exported for testing purposes
 export const paymentForAlice = purse.withdraw(
-  baytownBucks(10),
+  baytownBucks.make(10),
```
Comment (katelynsills) [inline examples/baytownBucks.js]: If we have `baytownBucks = amountMath.make`, this is the same as before.

File: examples/setup/initialPayments.js [hunk 5]
```diff
@@ -1,5 +1,4 @@
 // Start off Alice and Bob with a payment of some token
-export const makeInitialPayment = (mint, assay, allocation) => {
-  const purse = mint.mint(assay.makeUnits(allocation));
-  return purse.withdrawAll();
+export const makeInitialPayment = (mint, amountMath, allocation) => {
```
Comment (katelynsills) [inline examples/setup/initialPayments.js]: I wonder if we can get rid of this with the changes to ERTP making this helper unnecessary. We can certainly merge without changing this though.

**Comment by Chris-Hibbert** [comment]
https://github.com/Agoric/zoe-sketchbook/issues/4

File: examples/setup/store.js [hunk 6]
```diff
@@ -13,9 +14,9 @@ import { insist } from '@agoric/ertp/util/insist';
 function makeStore(keyName = 'key') {
   const wm = new Map();
   const insistKeyDoesNotExist = key =>
-    insist(!wm.has(key))([`${keyName} already registered`]);
+    assert(!wm.has(key), details`${key} already registered`);
```
Comment (katelynsills) [inline examples/setup/store.js]: Now that we have @agoric/store and @agoric/weak-store, can we use those instead? We can merge without that

File: examples/tradeWithAtomicSwap/alice.js [hunk 7]
```diff
@@ -18,37 +18,37 @@ const makeAlice = (zoe, installations, walletData) => {
...
-            units: simoleans(7),
-          },
-        ],
-        exitRule: {
-          kind: 'onDemand',
-        },
-      });
-      const payments = [wallet.withdraw('moola', moola(3)), undefined];
+      const payments = {
+        Price: simoleanIssuer.makeEmptyPurse().withdraw(simoleans(0)),
```
Comment (katelynsills) [inline examples/tradeWithAtomicSwap/alice.js]: We don't have to submit empty payments anymore, so we can leave this out!

File: examples/tradeWithAtomicSwap/bob.js [hunk 8]
```diff
@@ -1,66 +1,55 @@
...
+    assert(
+      installationHandle === installations.atomicSwap,
+      details`wrong installation`,
+    );
+    assert(issuerKeywordRecord.Asset === moolaIssuer, details`wrong issuer`);
+    assert(issuerKeywordRecord.Price === simoleanIssuer, details`wrong issuer`);
 
-    const payments = [undefined, wallet.withdraw('simolean', simoleans(7))];
+    const payments = {
+      Asset: moolaIssuer.makeEmptyPurse().withdraw(moola(0)),
```
Comment (katelynsills) [inline examples/tradeWithAtomicSwap/bob.js]: We can leave out this empty payment too

File: examples/zoe-contracts-copied/myFirstDapp.js [hunk 9]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-use-before-define */
```
Comment (katelynsills) [inline examples/zoe-contracts-copied/myFirstDapp.js]: Do we want to delete this one?

File: test/baytownBucks.test.js [hunk 10]
```diff
@@ -3,13 +3,19 @@ import { test } from 'tape-promise/tape';
...
 // correct balance.
 test('gets the balance of the payment for Alice', t => {
-  const unitOps = baytownBucksAssay.getUnitOps();
-  t.ok(unitOps.equals(paymentForAlice.getBalance(), baytownBucks(10)));
+  const { amountMath } = baytownBucksBundle;
+  const { issuer } = baytownBucksBundle;
+  t.ok(
+    amountMath.isEqual(
+      issuer.getAmountOf(paymentForAlice),
+      baytownBucks.make(10),
```
Comment (katelynsills) [inline test/baytownBucks.test.js]:
```suggestion
      baytownBucks(10),
```

File: test/tradeWithAtomicSwap.test.js [hunk 11]
```diff
@@ -14,43 +22,44 @@ import makeAlice from '../examples/tradeWithAtomicSwap/alice';
 test('perform trade using atomic swap contract', async t => {
   const { zoe, installations } = await setupZoe();
 
-  const inviteAssay = zoe.getInviteAssay();
+  const inviteissuer = zoe.getInviteIssuer();
```
Comment (katelynsills) [inline test/tradeWithAtomicSwap.test.js]:
```suggestion
  const inviteIssuer = zoe.getInviteIssuer();
```

File: test/tradeWithAtomicSwap.test.js [hunk 12]
```diff
@@ -14,43 +22,44 @@ import makeAlice from '../examples/tradeWithAtomicSwap/alice';
...
+  const moola = moolaIssuer.getAmountMath().make;
+  const simoleans = simoleanIssuer.getAmountMath().make;
 
   const registrar = makeRegistrar();
-  const inviteAssayRegKey = registrar.register('inviteAssay', inviteAssay);
-  const moolaAssayRegKey = registrar.register('moolaAssay', moolaAssay);
-  const simoleanAssayRegKey = registrar.register(
-    'simoleanAssay',
-    simoleanAssay,
+  const inviteissuerRegKey = registrar.register('inviteissuer', inviteissuer);
```
Comment (katelynsills) [inline test/tradeWithAtomicSwap.test.js]:
```suggestion
  const inviteissuerRegKey = registrar.register('inviteissuer', inviteIssuer);
```

**Review by katelynsills — CHANGES_REQUESTED** [review]
Some small changes. I think we can refactor some parts of this due to ERTP making things easier, but we can do that in a separate PR

File: examples/baytownBucks.js (hunk 1 above)
Comment (Chris-Hibbert) [reply examples/baytownBucks.js]: done

File: examples/baytownBucks.js (hunk 2 above)
Comment (Chris-Hibbert) [reply examples/baytownBucks.js]: done

File: examples/baytownBucks.js (hunk 3 above)
Comment (Chris-Hibbert) [reply examples/baytownBucks.js]: done

File: examples/baytownBucks.js (hunk 4 above)
Comment (Chris-Hibbert) [reply examples/baytownBucks.js]: done

File: examples/setup/initialPayments.js (hunk 5 above)
Comment (Chris-Hibbert) [reply examples/setup/initialPayments.js]: done

File: examples/setup/store.js (hunk 6 above)
Comment (Chris-Hibbert) [reply examples/setup/store.js]: done

File: examples/tradeWithAtomicSwap/alice.js (hunk 7 above)
Comment (Chris-Hibbert) [reply examples/tradeWithAtomicSwap/alice.js]: done

File: examples/tradeWithAtomicSwap/bob.js (hunk 8 above)
Comment (Chris-Hibbert) [reply examples/tradeWithAtomicSwap/bob.js]: done

File: examples/zoe-contracts-copied/myFirstDapp.js (hunk 9 above)
Comment (Chris-Hibbert) [reply examples/zoe-contracts-copied/myFirstDapp.js]: Yep, I agree.

File: test/tradeWithAtomicSwap.test.js (hunk 11 above)
Comment (Chris-Hibbert) [reply test/tradeWithAtomicSwap.test.js]: done

File: test/tradeWithAtomicSwap.test.js (hunk 12 above)
Comment (Chris-Hibbert) [reply test/tradeWithAtomicSwap.test.js]: done

**Review by katelynsills — APPROVED** [review]
Looks good to me! Thank you so much for updating this!

**Pull request merged**